rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -306,7 +315,12 @@ Candidates Candidates::make(
         assert(new_finish.has_value());
         initial_map.insert(
           {new_finish.value().finish_state().finish_time(),
-          Entry{i, new_finish.value().finish_state(), new_finish.value().wait_until()}});
+          Entry{
+            i,
+            new_finish.value().finish_state(),
+            new_finish.value().wait_until(),
```

Comment:
Here, the `wait_until` time of the entry with the charging task + original task is set to the time of the original task. I was wondering if that may be an issue in [this](https://github.com/osrf/rmf_core/blob/783c67a97c5e7220ec0a616d50b65b823513b2a1/rmf_task/src/rmf_task/agv/TaskPlanner.cpp#L884) line, where sometimes it might be more efficient to add a charging task much before we add the original task.

I think `expand_charger` should automatically take care of such cases, but I'm not a 100% sure.

Reply:
Yeah I think `expand_charger()` should account for such cases.